extensions/ql-vscode/src/common/vscode/vscode-app.ts

```diff
@@ -1,8 +1,24 @@
+import * as vscode from 'vscode';
 import { App } from '../app';
 import { AppEventEmitter } from '../events';
 import { VSCodeAppEventEmitter } from './events';
 
 export class ExtensionApp implements App {
+  public constructor(
+    public readonly extensionPath: string,
+    public readonly globalStoragePath: string,
+    public readonly workspaceStoragePath?: string
```

Comment:
Instead of making these properties, can we make them getters and save the extension context as a property instead? Something like this:

```ts
constructor(public readonly extensionContext: vscode.ExtensionContext) {}

public get extensionPath(): string {
  return this.extensionContext.extensionPath;
}

public get globalStoragePath(): string {
  return this.extensionContext.globalStorageUri.fsPath;
}

public get workspaceStoragePath(): string | undefined {
  return this.extensionContext.storageUri?.fsPath;
}
```

I think this will make it more resilient to changes in how VSCode determines the paths. Right now, they [seem to be computed](https://github.com/microsoft/vscode/blob/c9fd4761bf159592d1611d7a1b42fd5565120358/src/vs/workbench/api/common/extHostStoragePaths.ts#L41) at extension load and never change, but since [it's a getter](https://github.com/microsoft/vscode/blob/c9fd4761bf159592d1611d7a1b42fd5565120358/src/vs/workbench/api/common/extHostExtensionService.ts#L528), it could be changed to return a different value depending on changes we don't control.

Reply:
Thanks this is a good suggestion! I've made some changes.